##Trabajo final

###Isabella Palomino Yepez

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import statsmodels.api as sm           
import statsmodels.formula.api as smf

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [ ]:
df = pd.read_csv('recdemand.csv')
df

trips  quality  ski  income userfee  costC   costS   costH
0        0        0  yes       4      no  67.59  68.620  76.800
1        0        0   no       9      no  68.86  70.936  84.780
2        0        0  yes       5      no  58.12  59.465  72.110
3        0        0   no       2      no  15.79  13.750  23.680
4        0        0  yes       3      no  24.02  34.033  34.547
..     ...      ...  ...     ...     ...    ...     ...     ...
654     40        5  yes       5     yes  25.63  18.916  29.684
655     40        3  yes       3      no  18.46  13.330  25.800
656     40        2  yes       4      no  58.84  47.000  62.760
657     50        4  yes       4      no  35.67  25.694  37.269
658     88        2   no       2      no  22.49   6.210  25.460

[659 rows x 8 columns]

In [ ]:
reg = smf.ols('trips ~ quality + ski + income + userfee + costC + costS + costH', data = df)
res = reg.fit()
print(res.summary())

                            OLS Regression Results                            
Dep. Variable:                  trips   R-squared:                       0.279
Model:                            OLS   Adj. R-squared:                  0.271
Method:                 Least Squares   F-statistic:                     36.02
Date:                Fri, 22 Apr 2022   Prob (F-statistic):           1.29e-42
Time:                        13:07:09   Log-Likelihood:                -2038.8
No. Observations:                 659   AIC:                             4094.
Df Residuals:                     651   BIC:                             4130.
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          2.5615      0.572      4.

In [11]:
#Extrayendo los β estimados
res.params

Intercept         2.561541
ski[T.yes]        0.999160
userfee[T.yes]    8.280843
quality           0.913428
income           -0.207076
costC             0.063151
costS            -0.164493
costH             0.089516
dtype: float64

###El modelo con todas las variables introducidas como predictores tiene un  R2  de (0.279), es capaz de explicar el 27.9% de la variabilidad observada en el número de viajes en bote realizados en el lago Somerville. De acuerdo con el p-valor obtenido para el coeficiente parcial de regresión de las variables, el ingreso anual en miles de dólares (income) no es una variable significativa dado que (p-value 0.083>0.05). Asimismo, otra variable que no es significativa es el costo de visitar el lago Conroe (costC) ya que (p-value 0.66>0.05)

###Respecto a los β estimados, el coeficiente de la variable income (-0.207076) y de la variable costS(-0.164493), indican que ambas varibles disminuyen la probabilidad de viajar en bote en el lago Somerville. Por el contrario, aquellas variables con β estimados positivos como ski factor, income, userfee, costC y costH aumentan la propabilidad de viajar en bote en el lago Somerville.